From: Steve Dickson <steved@redhat.com>
Date: Fri, 26 Apr 2019 13:40:54 -0400
Subject: Removed a resource leak from mountd/fsloc.c
Origin: https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=26b84f73adc2ff9011bea307c47a4c0cdf4a3031
Bug-Debian: https://bugs.debian.org/513284

mountd/fsloc.c:97: overwrite_var: Overwriting "mp" in
	"mp = calloc(1UL, 16UL)" leaks the storage that "mp" points to.

Signed-off-by: Steve Dickson <steved@redhat.com>
---
 utils/mountd/fsloc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/utils/mountd/fsloc.c b/utils/mountd/fsloc.c
index bc737d191cf8..cf42944fa9a6 100644
--- a/utils/mountd/fsloc.c
+++ b/utils/mountd/fsloc.c
@@ -102,6 +102,7 @@ static struct servers *parse_list(char **list)
 		cp = strchr(list[i], '@');
 		if ((!cp) || list[i][0] != '/') {
 			xlog(L_WARNING, "invalid entry '%s'", list[i]);
+			free(mp);
 			continue; /* XXX Need better error handling */
 		}
 		res->h_mp[i] = mp;
-- 
2.27.0

